app/src/main/java/org/stepik/android/presentation/course_list_redux/dispatcher/CourseListActionDispatcher.kt

```diff
@@ -63,6 +65,31 @@ constructor(
                         onError = emptyOnErrorStub
                     )
             }
-        }
+
+            is CourseListFeature.Action.FetchCourseRecommendations -> {
+                compositeDisposable += Flowable
+                    .fromArray(SourceTypeComposition.CACHE, SourceTypeComposition.REMOTE)
+                    .concatMapSingle { sourceType ->
+                        courseRecommendationsInteractor.fetchCourseRecommendations()
```

Comment:
It seems, that `CourseRecommendationsInteractor` is not needed. Also `sourceType` should be used in recommendations query.